packages/styles/src/variables/components/_form-check.scss

```
@@ -14,7 +14,28 @@ $form-check-input-size: 1.375rem !default;
 $form-check-input-size-sm: spacing.$size-regular !default;
 $form-check-input-border-width: forms.$input-border-width !default;
 $form-check-input-focus-box-shadow: forms.$input-focus-box-shadow !default;
+$form-check-input-disabled-svg-border-width: $form-check-input-border-width * 2;
+$form-check-input-disabled-background-url-light: "data:image/svg+xml,%3csvg xmlns='http://www.w3.org/2000/svg'%3e%3crect width='100%25' height='100%25' fill='none' stroke='" +
+  color.$gray-60 + "' stroke-width='" + $form-check-input-disabled-svg-border-width
+  "' stroke-dasharray='4 2' stroke-vector-effect='non-scaling-stroke' linecap='square'/%3e%3c/svg%3e"; // Source: https://kovart.github.io/dashed-border-generator/ + added stroke-vector-effect
```

`linecap="square"` is for paths, not strokes, but if you add `stroke-linecap="square"` you have to calculate the width of the stroke within every visible dash value, so remove this or change it to its default, which is `stroke-linecap="butt"`.

The dasharray below works only for the exact size of the given checkbox, which is 22x22 pixel.

```suggestion
  "' stroke-dasharray='2 1.5 4 1.5 4 1.5 4 1.5 2 3.5 4 1.5 4 1.5 4 1.5 2 3.5 4 1.5 4 1.5 4 1.5 2 3.5 4 1.5 4 1.5 4' stroke-vector-effect='non-scaling-stroke' /%3e%3c/svg%3e"; // Source: https://kovart.github.io/dashed-border-generator/ + added stroke-vector-effect
```

---

@oliverschuerch I probably made a mistake when copy-paste or optimizing with svgo, I don't know. How did you generate this dasharray (just by curiosity)? We can make it work by using [`pathLength` attribute](https://developer.mozilla.org/en-US/docs/Web/SVG/Attribute/pathLength), but the visual space is quite full, so I made another pattern for the small variant

---

The new pattern looks great on a disabled checkbox, both small and large variant.
But for the radio button the outline is not dashed when disabled.

---

Oops on Chrome, the false value is interpreted as a 0 and infinity. Interesting.